### PR TITLE
Introduce "Interviewing" state

### PR DESCRIPTION
**Pull request by despo**
## Context

Introduces new interview state and transitions application_choice to that when an interview is created

## Changes proposed in this pull request

- Introduce new interview state
- Add audit entry 
- Amend filtered list sorting to include application choices in interview state in deadline approaching section
- Include interviewing in filtetered list
- Rename interviewing state via locale yaml so it appears the same as 'awaiting_provider_decision' to the candidate
- Display `interviewing` state as `awaiting_provider_decision` at Vendor API

## Link to Trello card
https://trello.com/c/uLndN1CF/3225-introduce-interviewing-state

## Things to check

- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-teacher-training#azure-hosting-devops-pipeline)

### CODE REVIEW

File: app/services/create_interview.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  )
+    @auth = ProviderAuthorisation.new(actor: actor)
+    @application_choice = application_choice
+    @provider = provider
+    @date_and_time = date_and_time
+    @location = location
+    @additional_details = additional_details
+  end
+
+  def call
```
Comment (AgaDufrat) [inline app/services/create_interview.rb:20]:
```suggestion
  def save
```

We tend to used `save` or `save!` for the services that change the state e.g. AcceptOffer WithdrawApplication DeclineOffer MakeAnOffer etc

File: spec/presenters/vendor_api/single_application_presenter_spec.rb [hunk 2]
```diff
@@ -585,4 +585,12 @@
       expect(presenter.as_json[:attributes][:recruited_at]).to be_present
     end
   end
+
+  describe 'attributes.status' do
+    it 'returns awaiting_provider_decision when status is interviewing' do
+      application_choice = build_stubbed(:application_choice, status: :interviewing)
```
Comment (benswannack) [inline spec/presenters/vendor_api/single_application_presenter_spec.rb:591]: It's probably worth adding a trait that sets up a realistic interviewing application with interviews etc.

File: config/locales/application_states.yml [hunk 3]
```diff
@@ -181,6 +191,12 @@ en:
       by: system
       description: Application choices that are unsubmitted are automatically rejected at the end of the cycle but can be carried over in the next cycle.
 
+    awaiting_provider_decision-interview:
+      name: Provider setups up an interview
```
Comment (benswannack) [inline config/locales/application_states.yml:195]:
```suggestion
      name: Provider sets up an interview
```

File: config/locales/application_states.yml [hunk 4]
```diff
@@ -181,6 +191,12 @@ en:
...
+      by: provider
+      description: |
+        Providers sets up an interview with candidates.
```
Comment (benswannack) [inline config/locales/application_states.yml:198]: We could put a bit more detail in this one, maybe something like this?
```suggestion
        Providers can set up an interview with candidates before making a decision
```

File: spec/presenters/vendor_api/single_application_presenter_spec.rb (hunk 2 above)
Comment (benswannack) [reply spec/presenters/vendor_api/single_application_presenter_spec.rb:591]: Oh – I see that's in the next PR 👍

**Review by benswannack — APPROVED** [review]
LGTM

**Pull request merged**